templates/terraform/resource.erb

```diff
@@ -296,24 +285,12 @@ func resource<%= resource_name -%>Update(d *schema.ResourceData, meta interface{
     d.Partial(true)
 
 <%  properties_by_custom_update(properties).each do |key, props| -%>
-<%  top_level_props = props.reject(&:flatten_object) + props.select(&:flatten_object).flat_map(&:properties) -%>
-    if <%= top_level_props.map { |prop| "d.HasChange(\"#{prop.name.underscore}\")" }.join ' || ' -%> {
+if <%= props.map { |prop| "d.HasChange(\"#{prop.name.underscore}\")" }.join ' || ' -%> {
```

Comment:
TODO - handle `d.HasChange` correctly. I'll probably either need to implement a `top_level_props` method on the Resource that is tolerant of multiple tiers of nesting or something more creative.